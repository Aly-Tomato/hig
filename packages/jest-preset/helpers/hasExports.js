module.exports = function hasExports(index, cases) {
  cases.forEach(({ name, value }) => {
    it(`exports ${name}`, () => {
      expect(index).toHaveProperty(name, value);
    });
  });
};
