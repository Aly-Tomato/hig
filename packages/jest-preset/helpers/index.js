/* eslint-disable global-require */
module.exports = {
  hasExports: require("./hasExports"),
  lastCallOfMock: require("./lastCallOfMock"),
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
