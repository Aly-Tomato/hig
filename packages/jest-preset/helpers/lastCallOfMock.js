module.exports = function lastCallOfMock(mockFn) {
  return mockFn.mock.calls[mockFn.mock.calls.length - 1];
};
