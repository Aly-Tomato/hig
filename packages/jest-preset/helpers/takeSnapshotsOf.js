const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const presenter = React.createElement(Component, props);
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
