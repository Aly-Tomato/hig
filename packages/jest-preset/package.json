{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "dependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.2",
    "enzyme": "^2.9.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.1",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.2"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
