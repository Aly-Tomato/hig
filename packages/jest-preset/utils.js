require("@hig/utils").generateId = jest.fn(key => `${key}-1`);
