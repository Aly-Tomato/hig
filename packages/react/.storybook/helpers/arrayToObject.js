export default function arrayToObject(arr) {
  var rv = {};
  for (var i = 0; i < arr.length; ++i) rv[arr[i]] = arr[i];
  return rv;
}
