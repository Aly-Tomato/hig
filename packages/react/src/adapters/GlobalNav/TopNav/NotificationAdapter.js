import React from "react";
import PropTypes from "prop-types";
import { Notification as VanillaNotification } from "hig-vanilla";

import HIGAdapter, {
  MountedByHIGParent,
  MapsPropToMethod,
  MountsAnyChild
} from "../../HIGAdapter";

function NotificationAdapter(props) {
  return (
    <HIGAdapter
      displayName="NotificationAdapter"
      HIGConstructor={VanillaNotification}
      {...props}
    >
      {adapterProps => (
        <div>
          <MountedByHIGParent mounter="addNotification" {...adapterProps} />
          <MapsPropToMethod
            value={props.timestamp}
            setter="setCreatedAt"
            {...adapterProps}
          />
          <MapsPropToMethod value={props.unread} {...adapterProps}>
            {(instance, value) =>
              value ? instance.markUnread() : instance.markRead()}
          </MapsPropToMethod>
          <MountsAnyChild mounter="setContent" {...adapterProps}>
            {props.children}
          </MountsAnyChild>
        </div>
      )}
    </HIGAdapter>
  );
}

NotificationAdapter.propTypes = {
  unread: PropTypes.bool,
  children: PropTypes.node,
  timestamp: PropTypes.string // ISO date string
};

export default NotificationAdapter;
