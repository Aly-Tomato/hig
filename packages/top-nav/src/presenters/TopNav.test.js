import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import TopNav from "./TopNav";

describe("top-nav/presenters/TopNav", () => {
  takeSnapshotsOf(TopNav, [
    {
      description: "renders with no props",
      props: {}
    },
    {
      description: "renders with actions and logo",
      props: {
        leftActions: <div className="leftActions" />,
        rightActions: <div className="rightActions" />,
        logo: <div className="logo" />
      }
    }
  ]);
});
