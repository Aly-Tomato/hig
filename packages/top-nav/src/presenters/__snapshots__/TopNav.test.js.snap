// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-nav/presenters/TopNav  1`] = `
<div
  className="hig__top-nav"
>
  <div
    className="hig__top-nav__item"
  >
    <div
      className="hig__top-nav__logo-wrapper"
    />
  </div>
  <div
    className="hig__top-nav__spacer"
  />
</div>
`;

exports[`top-nav/presenters/TopNav  2`] = `
<div
  className="hig__top-nav"
>
  <div
    className="leftActions"
  />
  <div
    className="hig__top-nav__item"
  >
    <div
      className="hig__top-nav__logo-wrapper"
    >
      <div
        className="logo"
      />
    </div>
  </div>
  <div
    className="hig__top-nav__spacer"
  />
  <div
    className="rightActions"
  />
</div>
`;
